enhancements/olm/catalogd-query-endpoint.md

```diff
@@ -0,0 +1,585 @@
+implementation that is proposed here.
+
+To keep this section succinct, document the details like API field
+changes, new images, and other implementation details in the
+**Implementation Details** section and record the reasons for not
+choosing alternatives in the **Alternatives** section at the end of
+the document.
+-->
+
+### Workflow Description
+
+To serve curated data from the FBC, a new HTTPS endpoint will be exposed by the existing service under the base URL as detailed in `API Specification`. The new endpoint will be derived from `.status.urls.base` following the pattern `/api/v1/query?[...parameters]` and will accept parameters which correspond to any of the fields of the `declcfg.Meta` catalog [atomic type](https://github.com/operator-framework/operator-registry/blob/e15668c933c03e229b6c80025fdadb040ab834e0/alpha/declcfg/declcfg.go#L111):
+
+```golang
+ type Meta struct {
+	Schema  string
+	Package string
+	Name	  string
+}
+```
+Query parameters will be logically ANDed and used to restrict response scope.   
+This API will be conditionally enabled by an upstream `APIV1QueryEndpoint` feature gate as part of a downstream `NewOLM{suffix}` style OCP TP feature gate, and will be disabled by default.
+
+<!--
+Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials.
+
+See
+https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow
+and https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md for more detailed examples.
+-->
+
+### API Extensions
+
+This proposal has no effect on OCP APIs.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No impacts, as Hypershift presently lacks support for catalogd.
+
+#### Standalone Clusters
+
+This proposal provides an additional HTTPS endpoint within an existing component service.
+
+#### Single-node Deployments or MicroShift
+
+This proposal provides an additional HTTPS endpoint within an existing component service.
```

Comment:
Microshift (and SNO for that matter) have really tight resource budgets, so I think it would be good to research those, write down what we think would be necessary to meet those standards, and get some SNO and microshift folks to review.